src/common/Cryptography/HMACSHA1.cpp

```diff
@@ -23,36 +23,61 @@
 
 HmacHash::HmacHash(uint32 len, uint8* seed)
 {
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L
+    m_ctx = HMAC_CTX_new();
+    HMAC_Init_ex(m_ctx, seed, len, EVP_sha1(), nullptr);
+#else
     HMAC_CTX_init(&m_ctx);
     HMAC_Init_ex(&m_ctx, seed, len, EVP_sha1(), nullptr);
+#endif
     memset(m_digest, 0, sizeof(m_digest));
 }
 
 HmacHash::~HmacHash()
 {
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L
+    HMAC_CTX_free(m_ctx);
+#else
     HMAC_CTX_cleanup(&m_ctx);
+#endif
 }
 
 void HmacHash::UpdateData(std::string const& str)
 {
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L
+    HMAC_Update(m_ctx, (uint8 const*)str.c_str(), str.length());
```

Comment:
most of the changes seems related to adding a "&" for the old version, maybe there is a better way to handle it than having the same preprocessor check "#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L" over and over across the whole file

Reply:
The intent of such a patch was to be a transitional, not a permanent solution, so it was a concious decision to not bend any class APIs to a single dependency's quirks. It is elaborate enough to not raise any questions about what happened there after some time passes, ridiculously easy to revert, easy to adapt to any particular OpenSSL API changes on per-case basis.
While it doesnt look pretty, there is no denying it, many FOSS projects who are in migration/transition to new OpenSSL versions chose the preprocessor solution to stick with.

Reply:
@jackpoz don't complain, after all we have this monster too dfd2660a85e4f0891c63009ee8425b2796586409